delivery_service.rb

```diff
@@ -0,0 +1,40 @@
+class DeliveryService
+  attr_accessor :cars_count, :bikes_count, :auto_park
+
+  require_relative 'car'
+  require_relative 'bike'
+  require_relative 'constants'
+
+  def initialize(cars_count = 6, bikes_count = 8)
+    @auto_park = []
+    cars_count.times do
+      @auto_park << Car.new
+    end
+    bikes_count.times do
+      @auto_park << Bike.new
+    end
+    @auto_park.sort_by! { |a| a.speed }
+  end
+
+  def find_transport(distance)
+    free = false
```

Comment:
You don't need to store availability status in here. I advice you to get rid of `output` method and to add `p "#{t.is_a?(Car) ? 'Car' : 'Bike'} is ready for delivery. Delivery time is approx #{t.delivery_time} minutes"` before you'll change the transport status. 

something like this:
```
transport = find_fitting_transport

if transport.present? 
  print your message that transport is ready and call some kind of start_delivery on this transport object
else
  raise 'No free transport'
end

private(also don't forget about incapsulation here)

def find_fitting_transport
  @auto_park.find { |a| a.available(because it has boolean value so it will in either way be false or 
  true) && a.distance >= distance } 
end

```

Reply:
Great! Thanks! Done.